### PR TITLE
[MAINTENANCE] Add tests for `_register_metric_functions` in the `MetricProvider` class hierarchy.

### PR DESCRIPTION
**Pull request by abegong**
Changes proposed in this pull request:
- Adds tests for `_register_metric_functions` to all the main `MetricProvider` classes: `MetricProvider`, `TableMetricProvider`, `ColumnMapMetricProvider`, `ColumnPairMapMetricProvider`, `MultiColumnMapMetricProvider`, and  `QueryMetricProvider`

### CODE REVIEW

File: tests/expectations/metrics/test_metric_providers.py [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+
+def test__multicolumn_map_metric__registration():
+    """This tests whether the MultiColumnMapMetricProvider class registers the correct metrics."""
+
+    registered_metric_keys = list(_registered_metrics.keys())
+    for key in registered_metric_keys:
+        assert "multicolumn_values.equal_seven" not in key
+    prev_registered_metric_key_count = len(registered_metric_keys)
+
+    class CustomColumnPairValuesEqualSeven(MulticolumnMapMetricProvider):
```
Comment (alexsherstinsky) [inline tests/expectations/metrics/test_metric_providers.py]: @abegong There seems to be a copy/paste typo in the name of the test class here (for illustrative purposes, it was probably intended to be named something like `CustomMulticolumnValuesEqualSeven`).  Thank you.

File: tests/expectations/metrics/test_metric_providers.py [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+
+        @multicolumn_condition_partial(engine=SqlAlchemyExecutionEngine)
+        def _sqlalchemy(cls, column_list, **kwargs):
+            raise NotImplementedError
+
+        @multicolumn_condition_partial(engine=SparkDFExecutionEngine)
+        def _spark(cls, column_list, **kwargs):
+            raise NotImplementedError
+
+    CustomColumnPairValuesEqualSeven()
```
Comment (alexsherstinsky) [inline tests/expectations/metrics/test_metric_providers.py]: @abegong Similarly, this was probably meant to be `CustomMulticolumnValuesEqualSeven()`.  Thank you.

**Review by alexsherstinsky — CHANGES_REQUESTED** [review]
@abegong Looks great (I just found what appears to be a small copy/paste typo).  Thank you.

**Comment by abegong** [comment]
@alexsherstinsky fixed it

**Review by alexsherstinsky — APPROVED** [review]
LGTM -- thank you!

**Pull request merged**